src/ControlSystem/IsSize.hpp

```diff
@@ -0,0 +1,22 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "ControlSystem/Systems/Size.hpp"
+#include "Domain/Structure/ObjectLabel.hpp"
+
+namespace control_system::size {
+// tt::is_a doesn't work because of domain::ObjectLabel and size_t
+template <template <domain::ObjectLabel, size_t> class U, typename T>
+struct is_size : std::false_type {};
```

Comment:
Do you need the first template parameter?  Looks like it's always the same, and it would make `is_size` and `is_size_v` consistent.